mcstasscript/interface/plotter.py

```diff
@@ -0,0 +1,650 @@
+                if data.plot_options.log:
+                    color_norm = matplotlib.colors.LogNorm(vmin=min_value,
+                                                           vmax=max_value)
+                    im = ax0.pcolormesh(x, y, to_plot,
+                                        cmap=cmap, norm=color_norm)
+                else:
+                    im = ax0.pcolormesh(x, y, to_plot, cmap=cmap, norm=norm)
+
+                # Add the colorbar
+                fig.colorbar(im, ax=ax0)
+
+                # Add a title
+                ax0.set_title(data.metadata.title)
+                
+                # Add axis labels
+                plt.xlabel(data.metadata.xlabel)
+                plt.ylabel(data.metadata.ylabel)
+
+                if data.plot_options.custom_ylim_top:
+                    ax0.set_ylim(top=data.plot_options.top_lim)
+
+                if data.plot_options.custom_ylim_bottom:
+                    ax0.set_ylim(bottom=data.plot_options.bottom_lim)
+
+                if data.plot_options.custom_xlim_left:
+                    ax0.set_xlim(left=data.plot_options.left_lim)
+
+                if data.plot_options.custom_xlim_right:
+                    ax0.set_xlim(right=data.plot_options.right_lim)
+
+            else:
+                print("Error, dimension not read correctly")
+
+        plt.show()
+
+
+class make_sub_plot:
+    """
+    make_plot plots contents of McStasData objects
+
+    Plotting is controlled through options assosciated with the
+    McStasData objects.  If a list is given, the plots appear in one
+    subplot.
+    """
+
+    def __init__(self, data_list, **kwargs):
+        """
+        plots McStasData, single object or list of McStasData
+
+        The options concerning plotting are stored with the data
+
+        Parameters
+        ----------
+        data_list : McStasData or list of McStasData
+            McStasData to be plotted
+        """
+        if not isinstance(data_list, McStasData):
+            print("number of elements in data list = "
+                  + str(len(data_list)))
+        else:
+            # Make list from single element to simplify syntax
+            data_list = [data_list]
+
+        number_of_plots = len(data_list)
+
+        # Relevant options:
+        #  select colormap
+        #  show / hide colorbar
+        #  custom title / label
+        #  color of 1d plot
+        #  overlay several 1d
+        #  log scale (o$rders of magnitude)
+        #  compare several 1d
+        #  compare 2D
+
+        # Find reasonable grid size for the number of plots
+        dim2 = math.ceil(math.sqrt(number_of_plots))
+        dim1 = math.ceil(number_of_plots/dim2)
+
+        if "fontsize" in kwargs:
+            plt.rcParams.update({'font.size': kwargs["fontsize"]})
+
+        fig, axs = plt.subplots(dim1, dim2, figsize=(13, 7))
+        axs = np.array(axs)
+        ax = axs.reshape(-1)
+
+        index = -1
+        for data in data_list:
```

Comment:
You could use `for index, data in enumerate(data_list):`
and then you don't need to initialise or increment `index` manually